Game/space-invaders/src/core/ParticleEmitter.h

```diff
@@ -0,0 +1,38 @@
+﻿#pragma once
+#include <memory>
+#include <vector>
+
+#include "GameObject.h"
+#include "SpriteRenderer.h"
+#include "Texture.h"
+#include "glm/glm.hpp"
+
+class ParticleEmitter
+{
+public:
+    ParticleEmitter(const char* spritePath);
+
+    void Update(float deltaTime);
+    void Render(const SpriteRenderer& renderer);
+    void Emit(const GameObject& source);
+    
+private:
+    struct ParticleProps
+    {
+        mutable GameObject Object;
```

Comment:
Not sure if we really need this GameObject, we could just use the Particle struct with addicional members. If in the future we add more complexity to the GameObject class, particles would propably don't need that complexity. But let's leave it like this for now.